src/util/test_utils.rs

```diff
@@ -0,0 +1,604 @@
+                        match parsed_sub_field {
+                            SubField::Padding(_) => {
+                                // The padding sub-field may be different on error
+                            }
+                            _ => {
+                                // Other sub-fields must be the same on error
+                                assert_eq!(
+                                    extra_field.0[i], parsed_sub_field.clone(),
+                                   "\nOn 'Err(_)'\noriginal: {:?}\nparsed:   {:?}\nfuzz_data: {:?}",
+                                   extra_field,
+                                   parsed_extra_field,
+                                   fuzz_data
+                                );
+                            }
+                        }
+                    }
+                    if add_padding == AddPadding::ToRear {
+                        panic!(
+                            "\nOn 'Err(_)', parsing a serialized ExtraField with padding at the rear may not fail\n({:?})\n({:?})\nfuzz_data: {:?}",
+                            extra_field,
+                            parsed_extra_field,
+                            fuzz_data,
+                        );
+                    }
+                }
+            };
+        }
+        Err(err) => {
+            panic!(
+                "Serializing an ExtraField may not fail\n({})\nextra field: {:?}\nfuzz_data: {:?}",
+                err, extra_field, fuzz_data
+            )
+        }
+    };
+
+    true
+}
+
+/// Fuzz helper function to create a raw extra field, called from the fuzz target
+pub fn fuzz_create_raw_extra_field(fuzz_data: &[u8]) -> Result<RawExtraField, String> {
+    match RawExtraField::try_from({
+        let add_padding = if fuzz_data.is_empty() {
+            AddPadding::ToMiddle
+        } else {
+            match fuzz_data.len() % 3 {
+                0 => AddPadding::ToFront,
+                1 => AddPadding::ToMiddle,
+                _ => AddPadding::ToRear,
+            }
+        };
+        fuzz_create_extra_field(fuzz_data, add_padding)
+    }) {
+        Ok(val) => Ok(val),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+/// Fuzz for transaction deserialization, called from the fuzz target
+pub fn fuzz_transaction_deserialize(fuzz_data: &[u8]) -> bool {
+    let fuzz_bytes = fuzz_data.to_vec();
+
+    // Transaction
+    if let Ok(val) = deserialize::<Transaction>(&fuzz_bytes[..]) {
+        assert_eq!(fuzz_bytes, serialize(&val));
+    }
+
+    let raw_extra_field = match fuzz_create_raw_extra_field(fuzz_data) {
+        Ok(val) => val,
+        Err(_) => {
+            // This may not fail, otherwise the test cannot continue
+            return true;
+        }
+    };
+
+    // let transaction = fuzz_create_transaction_1(fuzz_data, &raw_extra_field);
+    // let serialized_tx = serialize(&transaction);
+    // let _ = deserialize::<Transaction>(&serialized_tx[..]);
```

Comment:
O, that was meant to be added back in after I fixed some of the errors, but it can go back in. It caused the mini-fuzzing to err on `"called `Result::unwrap()` on an `Err` value: Custom { kind: Interrupted, error: ScriptNotSupported }"`, and commenting this out helped me to find other errors. I guess it can go back in as I have captured the errors with `fn previous_fuzz_transaction_deserialize_failures1()`, `fn previous_fuzz_transaction_deserialize_failures2()` and `fn previous_fuzz_transaction_hash_failures()`.